lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -109,6 +109,18 @@ def obfuscate_ip(ip, ips_dict)
         IPAddr.new(max_obfuscated + 1, Socket::AF_INET).to_s
       end
 
+      def hostname_match
+        bash_hostname = `uname -n`.chomp
+        foreman_hostname = ForemanRhCloud.foreman_host
```

Comment:
```suggestion
        foreman_hostname = ForemanRhCloud.foreman_host&.name
```
Otherwise the next `if` statement would always return `false`